<!DOCTYPE html>
<html lang="en">
  <head>
    <!-- Required meta tags always come first -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M" crossorigin="anonymous">
    <link rel="stylesheet" href="/css/design.css">
    <link rel="icon" href="/favicon.ico">
    <title>Detecting and correcting misclassified sequences in the large-scale public databases</title>
  </head>

  <body>
    <!-- Load jQuery before the content in order to support the papers page sorting -->
    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>

    <header>
  <div class="brand-bg-color">
    <div class="container">
      <div class="brand-navbar">
        <div class="row">
          <div class="flex-separate col-12 d-flex justify-content-between align-items-center">
            <a href="/" class="nav-link">
              <img src="/img/tulane-sse-logo.png" alt="Tulane School of Science and Engineering" id="header-logo"
                   style="height: 60px; margin: 10px 0; padding: 5px; border-radius: 4px;"/>
            </a>
            <!-- Navbar toggle button for mobile -->
            <button class="navbar-toggler d-block d-sm-none" type="button" data-toggle="collapse" data-target="#collapseNav" aria-controls="collapseNav" aria-expanded="false" aria-label="Toggle navigation">
              &#9776;
            </button>
            <!-- Desktop navigation links -->
            <div class="navbar-nav nav-horizontal d-none d-sm-flex">
              <a class="nav-item nav-link" href="/projects.html">Projects</a>
              <a class="nav-item nav-link" href="/papers/">Papers</a>
              <a class="nav-item nav-link" href="/grants/">Grants</a>
              <a class="nav-item nav-link" href="/news/">News</a>
              <a class="nav-item nav-link" href="/people.html">People</a>
              <a class="nav-item nav-link" href="/about/">About</a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Mobile navigation menu (only shows when collapsed) -->
  <div class="nav-bg-color d-sm-none">
    <nav class="container navbar navbar-dark navbar-expand-sm">
      <div class="collapse navbar-collapse" id="collapseNav">
        <div class="navbar-nav">
          <a class="nav-item nav-link" href="/projects.html">Projects</a>
          <a class="nav-item nav-link" href="/papers/">Papers</a>
          <a class="nav-item nav-link" href="/grants/">Grants</a>
          <a class="nav-item nav-link" href="/news/">News</a>
          <a class="nav-item nav-link" href="/people.html">People</a>
          <a class="nav-item nav-link" href="/about/">About</a>
        </div>
      </div>
    </nav>
  </div>
</header>

<main class="container">
  <br />
  <div class="row">
  <div class="col-12">
    <h2 class="page-title">Detecting and correcting misclassified sequences in the large-scale public databases</h2>
    

<h4>
By: 
Hamid Bagheri, Andrew J. Severin, and Hridesh Rajan

</h4>


<a href="annotations.pdf" class="icon">
  <img src="/img/pdf_icon.svg"/ alt="PDF Download">
  <span>Download Paper</span>
</a>

<br/>



<h4>Abstract</h4>
<p>
  <span id="Bagheri2020">Background: Creating a scalable computational infrastructure to analyze the wealth of information contained in data repositories is difficult due to significant barriers in organizing, extracting and analyzing relevant data. Shared data science infrastructures like Boag is needed to efficiently process and parse data contained in large data repositories. The main features of Boag are inspired from existing languages for data intensive computing and can easily integrate data from biological data repositories. Results: As a proof of concept, Boa for genomics, Boag, has been implemented to analyze RefSeq’s 153,848 annotation (GFF) and assembly (FASTA) file metadata. Boag provides a massive improvement from existing solutions like Python and MongoDB, by utilizing a domain-specific language that uses Hadoop infrastructure for a smaller storage footprint that scales well and requires fewer lines of code. We execute scripts through Boag to answer questions about the genomes in RefSeq. We identify the largest and smallest genomes deposited, explore exon frequencies for assemblies after 2016, identify the most commonly used bacterial genome assembly program, and address how animal genome assemblies have improved since 2016. Boag databases provide a significant reduction in required storage of the raw data and a significant speed up in its ability to query large datasets due to automated parallelization and distribution of Hadoop infrastructure during computations. Conclusions: In order to keep pace with our ability to produce biological data, innovative methods are required. The Shared Data Science Infrastructure, Boag, provides researchers a greater access to researchers to efficiently explore data in new ways. We demonstrate the potential of a the domain specific language Boag using the RefSeq database to explore how deposited genome assemblies and annotations are changing over time. This is a small example of how Boag could be used with large biological datasets.</span>
</p>





<h4>ACM Reference</h4>
<p>
<span id="Bagheri2020">Bagheri, H. et al. 2020. Detecting and correcting misclassified sequences in the large-scale public databases . <i>Oxford Bioinformatics</i>. 20, 436 (Aug. 2020). DOI:https://doi.org/10.1186/s12859-019-2967-2.</span>
</p>

<h4>BibTeX Reference</h4>
<pre><code>@article{Bagheri2020,
  author = {Hamid Bagheri and Andrew J. Severin and Hridesh Rajan },
  title = {Detecting and correcting misclassified sequences in the large-scale public databases },
  journal = {Oxford Bioinformatics},
  volume = {20},
  number = {436},
  article = {},
  month = {August},
  year = {2020},
  publisher = {Oxford},
  address = {},
  abstract = {Background: Creating a scalable computational infrastructure to analyze the wealth of information contained in data repositories is difficult due to significant barriers in organizing, extracting and analyzing relevant data. Shared data science infrastructures like Boag is needed to efficiently process and parse data contained in large data repositories. The main features of Boag are inspired from existing languages for data intensive computing and can easily integrate data from biological data repositories. Results: As a proof of concept, Boa for genomics, Boag, has been implemented to analyze RefSeq’s 153,848 annotation (GFF) and assembly (FASTA) file metadata. Boag provides a massive improvement from existing solutions like Python and MongoDB, by utilizing a domain-specific language that uses Hadoop infrastructure for a smaller storage footprint that scales well and requires fewer lines of code. We execute scripts through Boag to answer questions about the genomes in RefSeq. We identify the largest and smallest genomes deposited, explore exon frequencies for assemblies after 2016, identify the most commonly used bacterial genome assembly program, and address how animal genome assemblies have improved since 2016. Boag databases provide a significant reduction in required storage of the raw data and a significant speed up in its ability to query large datasets due to automated parallelization and distribution of Hadoop infrastructure during computations. Conclusions: In order to keep pace with our ability to produce biological data, innovative methods are required. The Shared Data Science Infrastructure, Boag, provides researchers a greater access to researchers to efficiently explore data in new ways. We demonstrate the potential of a the domain specific language Boag using the RefSeq database to explore how deposited genome assemblies and annotations are changing over time. This is a small example of how Boag could be used with large biological datasets.},
  doi={10.1186/s12859-019-2967-2},
}
</code></pre>


  </div>
  </div>
  <br />
</main>


<footer>
  <script>
    (function (i, s, o, g, r, a, m) {
      i['GoogleAnalyticsObject'] = r;
      i[r] = i[r] || function () {
        (i[r].q = i[r].q || []).push(arguments)
      },
      i[r].l = 1 * new Date();
      a = s.createElement(o),
      m = s.getElementsByTagName(o)[0];
      a.async = 1;
      a.src = g;
      m.parentNode.insertBefore(a, m)
    })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
    ga('create', 'UA-512574-3', 'auto');
    ga('send', 'pageview');
  </script>
  <div class="container">
    <div>
      <div class="row">
        <div class="col-12">
          <div class="row">
            <div class="col-xs-12 col-lg-3 tulane-logo-wrapper">
              <a href="http://www.iastate.edu">
                <img src="/img/tulane-logo.png" alt="Tulane University" class="tulane-logo">
              </a>
            </div>
            <div class="col-xs-12 col-sm-6 col-md-5 col-lg-4">
              <p>
                <b>Laboratory for Software Design</b>
              </p>
              <p>201 Lindy Claiborne Boggs Center</p>
              <p>6823 St. Charles Avenue</p>
              <p>New Orleans, LA 70118-5698</p>
              <p>Phone: (504) 865-5764</p>
              <p>E-Mail: hrajan@tulane.edu</p>
            </div>
            <div class="col-xs-12 col-sm-6 col-md-4 col-lg-4">
              <p>
              Copyright &copy;
              2025
              Tulane University. All rights reserved.
                <div class="d-none d-lg-block" style="margin-top: 0">
                  The research and educational activities described on these pages has been supported in part by the
                  <a href="http://www.nsf.gov">US National Science Foundation (NSF)</a>.
                </div>
              </p>
            </div>
            <div class="col-md-3 col-lg-1 d-none d-md-block project-logos">
              <div>
                <a href="http://web.cs.iastate.edu/~panini/">
                  <img src="/img/panini-logo.svg" class="logo" alt="Boa Logo">
                  Panini
                </div>
                <div>
                  <a href="http://boa.cs.iastate.edu">
                    <img src="/img/boa-logo.svg" class="logo" alt="Boa Logo">
                    Boa
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>



    <!-- jQuery first, then Tether, then Bootstrap JS. -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/tether/1.2.0/js/tether.min.js" integrity="sha384-Plbmg8JY28KFelvJVai01l8WyZzrYWG825m+cZ0eDDS1f7d/js6ikvy1+X+guPIB" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.11.0/umd/popper.min.js" integrity="sha384-b/U6ypiBEHpOf/4+1nzFpr53nxSS+GLCkfwBdFNTxtclqqenISfwAzpKaMNFNmj4" crossorigin="anonymous"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/js/bootstrap.min.js" integrity="sha384-h0AbiXch4ZDo7tp9hKZ4TsHbi047NrKGLO3SEJAg45jXxnGIfYzk4Si90RDIqNm1" crossorigin="anonymous"></script>
    <script src="/js/design.js"></script>
  </body>
</html>
